lib.mkEUI64Suffix: init

nukdokplex opened the pull request:
Added new function `lib.network.ipv6.mkEUI64Suffix` and created its alias in root: `lib.mkEUI64Suffix`.

This function can be useful in configuring IPv6 networking with Nix. It creates EUI-64 address suffix from 48-bit MAC address as described in docstring.

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nukdokplex committed:
lib.mkEUI64Suffix: init

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5706

rnhmjoj reviewed (approved):
The implementation seems correct to me and and doc string is good.
It could be useful to setup static addresses: flipping a bit in the address is quite annoying to do manually.

TheMaxMur reviewed (approved):
LGTM

nukdokplex committed:
lib.network: add lib.mkEUI64Suffix tests to network.sh

nukdokplex commented on lib/network/default.nix:
```diff
@@ -45,5 +45,39 @@ in
+      mac
+      : The MAC address (may contain these delimiters: `:`, `-` or `.` but it's not necessary)
+    */
+    mkEUI64Suffix =
```
@misuzu added some tests in network.sh

nukdokplex commented:
@infinisil @hsjobeki any thoughts on this?

hsjobeki reviewed (dismissed):
Nice work :+1: Does what it says, and adds value to lib.network

I found some slight adjustments:

- fix docstring typos, 
- tighten error message
- stricter delimiter handling would be nice

nukdokplex commented on lib/network/default.nix:
```diff
@@ -45,5 +53,49 @@ in
+      mac:
+      pipe mac [
+        # match mac address
+        (match "^([0-9A-Fa-f]{2})[-:.]?([0-9A-Fa-f]{2})[-:.]?([0-9A-Fa-f]{2})[-:.]?([0-9A-Fa-f]{2})[-:.]?([0-9A-Fa-f]{2})[-:.]?([0-9A-Fa-f]{2})$")
```
Very good idea for next pr, i would like to extract mac address parsing from here

nukdokplex commented on lib/tests/network.sh:
```diff
@@ -114,4 +114,42 @@ expectFailure '(internal._ipv6.split "/::/").prefixLength'   "is not a valid IPv
+expectSuccess 'mkEUI64Suffix "00.00.00.00.00.00"'  '"200:00ff:fe00:0000"'
+expectSuccess 'mkEUI64Suffix "000000000000"'       '"200:00ff:fe00:0000"'
+
+expectFailure 'mkEUI64Suffix "123456789AB"'        "doesn't meet MAC address criteria"
```
Totally agree

nukdokplex replied:
but not with testcase for mixed delimiters, i don't think it is good idea to require from future possible implementations of this function support of addresses with mixed delimiters. this is just an undocumented feature which shouldn't hurt anyone

rnhmjoj commented on lib/network/default.nix:
```diff
@@ -45,5 +53,49 @@ in
+      # Argumemts
+
+      mac
+      : The MAC address (may contain these delimiters: `:`, `-` or `.` but it's not necessary)
```
Well, it's ok to keep them. I just think it's strange: literally no option in NixOS uses these others formats

nukdokplex committed:
lib.mkEUI64Suffix: bring @hsjobeki suggestions

renamed `mac` to `octets` in hextets combining step for better var
naming; rephrased error message to provide a hint of expected format;
replaced `Arguments` with `Inputs` in docstring; added more test cases
for invalid hex digits; added comments in hextets combining step;

nukdokplex committed:
lib.mkEUI64Suffix: fix syntax errors and tests
